src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -751,7 +751,7 @@ private bool FailsConstraintChecks(MethodSymbol method, out ArrayBuilder<TypePar
             }
             else if (containingTypeMapOpt == null)
             {
-                if (MemberGroupContainsOverride(members, member))
+                if (MemberGroupContainsMoreDerivedOverride(members, member, checkOverrideContainingType: true, ref useSiteDiagnostics))
```

Comment:
Notes to self from chat with Neal:

The spec says that when collecting candidates for a method group, we should only keep the definition (and discard the overrides). But for practical reasons, the compiler instead keeps the most derived override instead (and we fix it up when emitting).

Previously, when comparing two candidates, we would see if one overrides the other (by looking up the overridden member chain). If they do, we keep the more derived one. But that doesn't work well in multi-assembly scenarios with particular build sequences (two overrides may not belong to the same chain).

Now, when comparing two candidates, we look all the way up their overridden member chains to find the definition, and if the definitions match, then we keep the more derived one (by looking at containing types).